plugin/secrets_service_account_key.go

```diff
@@ -232,3 +403,84 @@ would generate service account keys for the "deploy" role set.
 On the backend, each roleset is associated with a service account under
 which secrets/keys are created.
 `
+
+type serviceAccountKeyCacheCollection struct {
+	Items map[string]*serviceAccountKeyCacheItem
+}
+
+func newServiceAccountKeyCacheCollection(item *serviceAccountKeyCacheItem, itemKey string) *serviceAccountKeyCacheCollection {
```

Comment:
This function does too many things. I think it would be better to split this to:
  - `newServiceAccountKeyCacheCollection() *serviceAccountKeyCacheCollection` and
  - `(c *serviceAccountKeyCacheCollection) putItem(itemKey string, item *serviceAccountKeyCacheItem) err`

Reply:
Noted, I'll update it in the next commit.